### PR TITLE
Added replace image registry with harbor

### LINKED ISSUE #882: [Sample] rewrite image registry
### Problem Statement

rewrite the image registry, to force all pulls throug a proxy cache e.g. with harbor

### Solution Description

```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: replace-image-registry
spec:
  background: false
  rules:
    - name: replace-image-registry
      match:
        any:
        - resources:
            kinds:
            - Pod
      mutate:
        foreach:
        - list: "request.object.spec.containers"
          context: 
          - name: imageData
            imageRegistry: 
              reference: "{{ element.image }}"
          preconditions:
            all:
            - key: "{{imageData.image}}"
              operator: Equals
              value: "docker.io/*"
          patchStrategicMerge:
            spec:
              containers:
              - name: "{{ element.name }}"
                image: "harbor2.zoller.com/dockerhub-proxy/{{imageData.repository}}:{{imageData.identifier}}"

```

### Example "Good" Resource

_No response_

### Example "Bad" Resource

_No response_

### Other Comments

_No response_

### Slack discussion

https://kubernetes.slack.com/archives/CLGR9BJU9/p1704740196988469

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have verified the policy does not exist in the [samples library](https://kyverno.io/policies/).

### PR DESCRIPTION
**Pull request by siddhikhapare**
## Related Issue(s)
Closes #882 

## Description

* Adds a new sample policy to rewrite images from docker.io to be directed to Harbor's pull-through cache.
* Adds the `--registry` flag to current CI testing with the Kyverno CLI to enable testing of this policy since it uses image registry context data.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by siddhikhapare** [comment]
@chipzoller Please review my PR.

File: other/replace-image-registry-with-harbor/replace-image-registry-with-harbor.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      https://kyverno.io/docs/writing-policies/external-data-sources/#variables-from-image-registries
+spec:
+  background: false
+  rules:
+    - name: replace-image-registry-with-harbor-pod-containers
+      match:
+        any:
+          - resources:
+            kinds:
+              - Pod
```
Comment (chipzoller) [inline other/replace-image-registry-with-harbor/replace-image-registry-with-harbor.yaml:31]: Indentation is wrong here

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: other/replace-image-registry-with-harbor/replace-image-registry-with-harbor.yaml (hunk 1 above)
Comment (siddhikhapare) [reply other/replace-image-registry-with-harbor/replace-image-registry-with-harbor.yaml:31]: I will update it. Thanks.

**Comment by chipzoller** [comment]
Significant changes were need here so I have taken this over.

**Comment by chipzoller** [comment]
Need to fix a couple of CLI tests now the `--registry` flag has been added. Will do shortly.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**